libmamba/src/core/satisfiability_error.cpp

```diff
@@ -0,0 +1,348 @@
+                auto source = problem.source();
+                auto target = problem.target();
+                auto dep = problem.dep();
+                auto type = map_problem(problem.type);
+                switch (problem.type)
+                {
+                    case SOLVER_RULE_PKG_CONSTRAINS:
+                    {
+                        // TODO
+                        if (!source || !target || !dep)
+                        {
+                            warn_unexpected_problem(problem);
+                            break;
+                        }
+                        auto src_id
+                            = ensure_solvable(problem.source_id, std::move(source).value(), type);
+                        auto tgt_id
+                            = ensure_solvable(problem.target_id, std::move(target).value(), type);
+                        auto const edge = ConstraintEdge{ DependencyInfo(dep.value()) };
+                        add_expanded_deps_edges(src_id, problem.dep_id, edge);
+                        add_conflict(src_id, tgt_id);
+                        break;
+                    }
+                    case SOLVER_RULE_PKG_REQUIRES:
+                    {
+                        if (!dep || !source)
+                        {
+                            warn_unexpected_problem(problem);
+                            break;
+                        }
+                        auto src_id
+                            = ensure_solvable(problem.source_id, std::move(source).value(), type);
+                        auto edge = RequireEdge{ DependencyInfo(dep.value()) };
+                        add_expanded_deps_edges(src_id, problem.dep_id, edge);
+                        break;
+                    }
+                    case SOLVER_RULE_JOB:
+                    case SOLVER_RULE_PKG:
+                    {
+                        if (!dep)
+                        {
+                            warn_unexpected_problem(problem);
+                            break;
+                        }
+                        // TODO check here that depId exists
+                        auto edge = RequireEdge{ DependencyInfo(dep.value()) };
+                        add_expanded_deps_edges(m_root_node, problem.dep_id, edge);
+                        break;
+                    }
+                    case SOLVER_RULE_JOB_NOTHING_PROVIDES_DEP:
+                    case SOLVER_RULE_PKG_NOTHING_PROVIDES_DEP:
+                    case SOLVER_RULE_JOB_UNKNOWN_PACKAGE:
+                    {
+                        if (!source || !dep)
+                        {
+                            warn_unexpected_problem(problem);
+                            break;
+                        }
+                        auto edge = RequireEdge{ DependencyInfo(dep.value()) };
+                        auto src_id
+                            = ensure_solvable(problem.source_id, std::move(source).value(), type);
```

Comment:
why do we set the problem type of the src_id here as well ? 
I expect to have the problem type here of the format <source=existing_solvable> requires <target=non_existing_package> so I would only set the problem type for the target package.